src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleKey.java

```diff
@@ -83,15 +75,45 @@ public final String toString() {
   }
 
   /** Returns the canonical name of the repo backing this module. */
-  public RepositoryName getCanonicalRepoName() {
+  public RepositoryName getCanonicalRepoName(boolean hasUniqueVersion) {
     if (WELL_KNOWN_MODULES.containsKey(getName())) {
       return WELL_KNOWN_MODULES.get(getName());
     }
     if (ROOT.equals(this)) {
       return RepositoryName.MAIN;
     }
-    return RepositoryName.createUnvalidated(
-        String.format("%s~%s", getName(), getVersion().isEmpty() ? "override" : getVersion()));
+    String suffix;
+    if (hasUniqueVersion) {
+      // If there is a unique version of this module in the entire dep graph, we elide the version
```

Comment:
Can you please update the doc here as well: https://bazel.build/external/module#repository_names_and_strict_deps?